File: python/problem085.py
def count_rectangles(x, y):
    count = 0
    for i in range(1, x + 1):
        for j in range(1, y + 1):
            count += (x + 1 - i) * (y + 1 - j)
    return count

nearest_x = 0
nearest_y = 0
nearest_value = 2000000

for a in range(1, 100):
    for b in range(1, 100):
        value = count_rectangles(a, b) - 2000000
        if value < 0:
            value *= -1
        if value < nearest_value:
            nearest_value = value
            nearest_x = a
            nearest_y = b

result = nearest_x * nearest_y
print(result)

File: javascript/problem085.js
function countRectangles(x, y){
  var count = 0;
  for(i=1; i<=x; i++){
    for(j=1; j<=y; j++){
      count = count + (x + 1 - i)*(y + 1 - j);
    }
  }
  return count;
}

var nearestX = 0;
var nearestY = 0;
var nearestValue = 2000000;

for(a=1; a<100; a++){
  for(b=1; b<100; b++){
    var value = countRectangles(a, b) - 2000000;
    if(value < 0){
      value = value * (-1);
    }
    if(value < nearestValue){
      nearestValue = value;
      nearestX = a;
      nearestY = b;
    }
  }
}

var result = nearestX * nearestY;

console.log(result);
